### PR TITLE
Add SQL::Relation#import

### PR DESCRIPTION
**Pull request by flash-gordon**
This method import data from another relation. In common cases, it will use the `INSERT ... SELECT ...` statement which is known as being very effective.

### CODE REVIEW

File: lib/rom/sql/relation/writing.rb [hunk 1]
```diff
@@ -77,6 +77,44 @@ def update(*args, &block)
         def delete(*args, &block)
           dataset.delete(*args, &block)
         end
+
+        # Insert tuples from other relation
+        # NOTE: The method is implicitly uses a transaction
```
Comment (cflipse) [inline lib/rom/sql/relation/writing.rb]: ~is~

File: lib/rom/sql/relation/writing.rb [hunk 2]
```diff
@@ -77,6 +77,44 @@ def update(*args, &block)
...
+        #
+        # @example
+        #   users.insert_from(new_users)
```
Comment (cflipse) [inline lib/rom/sql/relation/writing.rb]: should update to import here.

File: lib/rom/sql/relation/writing.rb [hunk 3]
```diff
@@ -77,6 +77,44 @@ def update(*args, &block)
...
+        #   @option [Integer] :slice
+        #     Split loading into batches of provided size,
+        #     every batch will be processed in a separate
+        #     transaction block
+        #
+        # @overload import(other, options)
+        #   Import data from another relation. The source
+        #   relation will be materialized before loading
+        #
+        #   @params [Relation] other_sql_relation
```
Comment (cflipse) [inline lib/rom/sql/relation/writing.rb]: is other necessarily an SQL relation here?

**Comment by flash-gordon** [comment]
@cflipse thanks for the review ❤️

**Pull request merged**